src/shogun/features/DotFeatures.cpp

```diff
@@ -63,7 +63,7 @@ void CDotFeatures::dense_dot_range(float64_t* output, int32_t start, int32_t sto
 
 	int32_t num_threads;
 	int32_t step;
-	auto pb = progress(range(num_vectors), *this->io);
+	auto pb = SG_PROGRESS(range(num_vectors), *this->io);
```

Comment:
why the this->io? can't we do this via the macro as well?

Reply:
we dont always pass *this->io so i guess its better to pass it when we need to